cli/src/main/java/de/jplag/cli/JPlagVersionChecker.java

```diff
@@ -0,0 +1,84 @@
+package de.jplag.cli;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Optional;
+
+import javax.json.Json;
+import javax.json.JsonArray;
+import javax.json.JsonReader;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.jplag.JPlag;
+import de.jplag.reporting.reportobject.model.Version;
+
+/**
+ * Handles the check for newer versions.
+ */
+public class JPlagVersionChecker {
+    private static final String API_URL = "https://api.github.com/repos/jplag/JPlag/releases";
+    private static final Logger logger = LoggerFactory.getLogger(JPlagVersionChecker.class);
+    private static final String WARNING_UNABLE_TO_FETCH = "Unable to fetch version information. New version notification will not work.";
+    private static final String NEWER_VERSION_AVAILABLE = "There is a newer version ({}) available. You can fetch the newest version here: https://github.com/jplag/JPlag/releases.";
+
+    private JPlagVersionChecker() {
+
+    }
+
+    /**
+     * Prints a warning if a newer version is available on GitHub.
+     */
+    public static void printVersionNotification() {
+        Optional<Version> newerVersion = checkForNewVersion();
+        newerVersion.ifPresent(version -> logger.warn(NEWER_VERSION_AVAILABLE, version));
+    }
+
+    private static Optional<Version> checkForNewVersion() {
+        try {
+            JsonArray array = fetchApi();
+            Version newest = getNewestVersion(array);
+            Version current = JPlag.JPLAG_VERSION;
+
+            if (newest.compareTo(current) > 0) {
+                return Optional.of(newest);
+            }
+        } catch (IOException | URISyntaxException e) {
```

Comment:
This call should never fail. Maybe catch more in this case. You don't want to crash JPlag because of an unknown scenario. E.g., Server returns 502 -> I don't know the exception here .. Or JSON invalid -> Jackson will throw an exception ..

Reply:
In both cases, no warning or error, just to debug the log. If the user is the a train, he has no connection, then it should not fail as this is technically not a unexpected failure but a scenario that can be expected. Also if the latter case is true, JPlag still works as intended.